src/main/java/org/isf/medicalstockward/manager/MovWardBrowserManager.java

```diff
@@ -296,4 +297,61 @@ public int compare(MovementForPrint o1, MovementForPrint o2) {
 			}
 		}
 	}
+
+	/**
+	 * Delete a {@link MovementWard}
+	 *
+	 * @param movWardToDelete - the movement ward
+	 * @throws OHServiceException if an error occurs retrieving the medical.
+	 */
+	@Transactional(rollbackFor = OHServiceException.class)
+	public void deleteLastMovementWard(MovementWard movWardToDelete) throws OHServiceException {
+		MovementWard lastMovementWard = ioOperations.getLastMovementWard(movWardToDelete.getWard());
+		if (lastMovementWard.getCode() != movWardToDelete.getCode()) {
+			throw new OHDataValidationException(
+							new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.onlythelastmovementcanbedeleted.msg")));
+		}
+		Ward wardTo = movWardToDelete.getWardTo();
+		Medical medical = movWardToDelete.getMedical();
+		Lot lot = movWardToDelete.getLot();
+		if (movWardToDelete.getWardTo() != null) {
+			List<MovementWard> latestMovementWardList = ioOperations.getMovementWardByWardMedicalAndLotAfterOrSameDate(wardTo.getCode(), medical.getCode(), lot.getCode(), movWardToDelete.getDate());
+			if (latestMovementWardList.size() == 1) {
+				MovementWard lastMovInWardTo = ioOperations.getLastMovementWard(wardTo);
+				Ward lastwardTo = lastMovInWardTo.getWardTo();
+				Medical medicalLastWardTo = lastMovInWardTo.getMedical();
+				Lot lotLastWardTo = lastMovInWardTo.getLot();
+				MedicalWard medWard = this.getMedicalWardByWardMedicalAndLot(lastwardTo.getCode(), medicalLastWardTo.getCode(), lotLastWardTo.getCode());
```

Comment:
There's no need to extract again, the parameters are the ones extracted before. Moreover, you'll get a NPE on the `lastwardTo.getCode()` of the `lastMovInWardTo` which of course has not a destination ward but only an origin one.

```suggestion
				MedicalWard medWard = getMedicalWardByWardMedicalAndLot(wardTo.getCode(), medical.getCode(), lot.getCode());
```